core/commands/store/hosts/hosts.go

```diff
@@ -104,15 +108,39 @@ Mode options include:` + hub.AllModeHelpText,
 		if err != nil {
 			return err
 		}
-
 		return SyncHosts(req.Context, n, mode)
 	},
 }
 
 func SyncHosts(ctx context.Context, node *core.IpfsNode, mode string) error {
+	api, err := coreapi.NewCoreAPI(node)
+	if err != nil {
+		return err
+	}
 	nodes, err := hub.QueryHosts(ctx, node, mode)
 	if err != nil {
 		return err
 	}
-	return storage.SaveHostsIntoDatastore(ctx, node, mode, nodes)
+	var wg sync.WaitGroup
+	var mu sync.Mutex
+	hs := make([]*hubpb.Host, 0)
+	for _, h := range nodes {
+		wg.Add(1)
+		go func(h *hubpb.Host) {
+			if err := api.Swarm().Connect(ctx, peer.AddrInfo{ID: peer.ID(h.NodeId)}); err != nil {
+				mu.Lock()
+				// push back
+				hs = append(hs, h)
+				mu.Unlock()
+			} else {
+				mu.Lock()
+				// push front
+				hs = append([]*hubpb.Host{h}, hs...)
```

Comment:
It should prepend in the order of the host nodes. Host nodes are ordered by score, if you push front every time, it will reverse the scoring.

Reply:
ah, i c